### PR TITLE
Use upstream `StatisticsConverter` from arrow-rs in DataFusion

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/10922
Closes https://github.com/apache/datafusion/issues/11000

## Rationale for this change

Now that @efredine  has ported this code to arrow in https://github.com/apache/arrow-rs/pull/6046 we can remove the copy in DataFusion

## What changes are included in this PR?

Remove code from DataFusion and use the upstream arrow-rs version

### CODE REVIEW

File: datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs [hunk 1]
```diff
@@ -356,20 +356,24 @@ impl<'a> RowGroupPruningStatistics<'a> {
         &'a self,
         column: &'b Column,
     ) -> Result<StatisticsConverter<'a>> {
-        StatisticsConverter::try_new(&column.name, self.arrow_schema, self.parquet_schema)
+        Ok(StatisticsConverter::try_new(
```
Comment (alamb) [inline datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs:359]: this is required to get the errors to convert

File: datafusion-examples/examples/parquet_index.rs [hunk 2]
```diff
@@ -25,12 +25,10 @@ use arrow_schema::SchemaRef;
...
 use datafusion::catalog::Session;
 use datafusion::datasource::listing::PartitionedFile;
-use datafusion::datasource::physical_plan::{
-    parquet::StatisticsConverter,
-    {FileScanConfig, ParquetExec},
-};
+use datafusion::datasource::physical_plan::{FileScanConfig, ParquetExec};
 use datafusion::datasource::TableProvider;
 use datafusion::execution::object_store::ObjectStoreUrl;
+use datafusion::parquet::arrow::arrow_reader::statistics::StatisticsConverter;
```
Comment (alamb) [inline datafusion-examples/examples/parquet_index.rs:31]: code has been ported to arrow-rs (🙏  @efredine )

File: datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs [hunk 3]
```diff
@@ -356,20 +356,24 @@ impl<'a> RowGroupPruningStatistics<'a> {
...
+            self.parquet_schema,
+        )?)
     }
 }
 
 impl<'a> PruningStatistics for RowGroupPruningStatistics<'a> {
     fn min_values(&self, column: &Column) -> Option<ArrayRef> {
         self.statistics_converter(column)
-            .and_then(|c| c.row_group_mins(self.metadata_iter()))
+            .and_then(|c| Ok(c.row_group_mins(self.metadata_iter())?))
```
Comment (alamb) [inline datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs:370]: this is necessary to convert from `ArrowError` to `DataFusionError`

File: datafusion/core/benches/parquet_statistic.rs [hunk 4]
```diff
@@ -1,287 +0,0 @@
-// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (alamb) [inline datafusion/core/benches/parquet_statistic.rs:1]: moved to arrow

**Comment by alamb** [comment]
@Ted-Jiang  I wonder if you have a moment to review this PR?

File: datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs (hunk 3 above)
Comment (comphead) [reply datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs:370]: can we use map_err? or into?

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb and @efredine for porting this

**Review by Ted-Jiang — APPROVED** [review]
Make sense to me.

**Comment by alamb** [comment]
Thank you for the reviews @Ted-Jiang and @comphead

**Pull request merged**